src/main/java/org/opensearch/ad/ml/ModelManager.java

```diff
@@ -470,51 +437,34 @@ public void stopModel(String detectorId, String modelId) {
      */
     public void stopModel(String detectorId, String modelId, ActionListener<Void> listener) {
         logger.info(String.format(Locale.ROOT, "Stopping detector %s model %s", detectorId, modelId));
-        stopModel(
-            forests,
-            modelId,
-            this::toCheckpoint,
-            ActionListener.wrap(r -> stopModel(thresholds, modelId, this::toCheckpoint, listener), listener::onFailure)
-        );
+        stopModel(forests, modelId, ActionListener.wrap(r -> stopModel(thresholds, modelId, listener), listener::onFailure));
     }
 
-    private <T> void stopModel(
-        Map<String, ModelState<T>> models,
-        String modelId,
-        Function<T, String> toCheckpoint,
-        ActionListener<Void> listener
-    ) {
+    private <T> void stopModel(Map<String, ModelState<T>> models, String modelId, ActionListener<Void> listener) {
         Instant now = clock.instant();
         Optional<ModelState<T>> modelState = Optional
             .ofNullable(models.remove(modelId))
             .filter(model -> model.getLastCheckpointTime().plus(checkpointInterval).isBefore(now));
         if (modelState.isPresent()) {
-            modelState
-                .ifPresent(
-                    model -> checkpointDao
-                        .putModelCheckpoint(
-                            modelId,
-                            toCheckpoint.apply(model.getModel()),
-                            ActionListener.wrap(r -> listener.onResponse(null), listener::onFailure)
-                        )
-                );
+            T model = modelState.get().getModel();
+            if (model instanceof RandomCutForest) {
+                checkpointDao
+                    .putRCFCheckpoint(
+                        modelId,
+                        (RandomCutForest) model,
+                        ActionListener.wrap(r -> listener.onResponse(null), listener::onFailure)
+                    );
+            } else if (model instanceof ThresholdingModel) {
+                checkpointDao
+                    .putThresholdCheckpoint(
+                        modelId,
+                        (ThresholdingModel) model,
+                        ActionListener.wrap(r -> listener.onResponse(null), listener::onFailure)
+                    );
+            }
```

Comment:
If we use "else if" in line 457, should we throw IllegalStateException after line 464? Alternatively, 457 could just be else.

Reply:
I'll throw IllegalStateException after line 464